runtime/lib/backend/lightning/lightning_kokkos/LightningKokkosSimulator.cpp

```diff
@@ -12,14 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <Kokkos_Complex.hpp>
+#include <Kokkos_Core.hpp>
+
 #include "LightningKokkosSimulator.hpp"
 
 namespace Catalyst::Runtime::Simulator {
 
 auto LightningKokkosSimulator::AllocateQubit() -> QubitIdType
 {
     const size_t num_qubits = this->device_sv->getNumQubits();
-    this->device_sv = std::make_unique<StateVectorT>(num_qubits + 1);
+
+    if (!num_qubits) {
+        this->device_sv = std::make_unique<StateVectorT>(1);
+        return this->qubit_manager.Allocate(num_qubits);
+    }
+
+    std::vector<Kokkos::complex<double>> data = this->device_sv->getDataVector();
+    const size_t dsize = data.size();
+    data.resize(dsize << 1UL);
+
+    auto src = data.begin();
+    std::advance(src, dsize - 1);
+
+    for (auto dst = data.end() - 2; src != data.begin();
+         std::advance(src, -1), std::advance(dst, -2)) {
+        *dst = std::move(*src);
+        *src = Kokkos::complex<double>(.0, .0);
+    }
+
+    this->device_sv = std::make_unique<StateVectorT>(data);
```

Comment:
I don't think this will be GPU compatible (unless you are happy to make the copy of the device -> host with the call to `getDataVector()`). You may want to adapt this to instead use Kokkos Views and such.

Reply:
After supporting the dynamic qubit alloc/dealloc in the C++ API of lightning simulators, this logic will be removed from the runtime. we have more control over the managed host-device memory for GPU devices :crossed_fingers:, although I'm not sure if we necessarily want the dynamic qubit feature for LightningGPU and LightningKokkos (w/ CUDA or HIP kernels) because of obviously performance concerns for a foreseeable future. I should also add that by developing a tensor-network simulator (LighntingTensorNet :P maybe!), we no longer need these patches for Lightning state-vector simulators necessarily! Instead, we can say that dynamic qubit management is only available by LighntingTensorNet ;)

Reply:
For now this inefficiency is acceptable as it's a rarely used feature. We'll revisit if it comes up again :)